From stern@rowland.harvard.edu  Sat Sep  8 07:45:59 2007
From: Alan Stern <stern@rowland.harvard.edu>
Date: Tue, 4 Sep 2007 09:53:24 -0400 (EDT)
Subject: USB: fix location of statement label in dummy-hcd
To: Greg KH <greg@kroah.com>
Cc: USB development list <linux-usb-devel@lists.sourceforge.net>
Message-ID: <Pine.LNX.4.44L0.0709040952400.5120-100000@iolanthe.rowland.org>


This patch (as984) fixes a rather elementary mistake in dummy_hcd.
The new statement label should come before the spin_unlock_irqrestore,
not after it.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/dummy_hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/dummy_hcd.c
+++ b/drivers/usb/gadget/dummy_hcd.c
@@ -1001,8 +1001,8 @@ static int dummy_urb_enqueue (
 	if (!timer_pending (&dum->timer))
 		mod_timer (&dum->timer, jiffies + 1);
 
-	spin_unlock_irqrestore (&dum->lock, flags);
  done:
+	spin_unlock_irqrestore(&dum->lock, flags);
 	return rc;
 }
 
